File: src/main/java/com/visited/visited/controller/BoardController.java
package com.visited.visited.controller;

import com.visited.visited.model.req.board.InsertBoardDto;
import com.visited.visited.model.res.Response;
import com.visited.visited.model.res.board.SimpleBoardDto;
import com.visited.visited.service.BoardService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@Controller
@RequiredArgsConstructor
@RequestMapping("board")
public class BoardController {
	private final BoardService boardService;
	@PostMapping
	@ResponseBody
	public Response insert(@RequestBody InsertBoardDto insertBoardDto){
		boardService.insertBoard(insertBoardDto);

		return new Response();
	}
	@GetMapping
	public String mainPage(Model model){
		List<SimpleBoardDto> boards = boardService.getBoards();
		model.addAttribute("boards", boards);
		return "main";
	}
}

File: api/src/main/kotlin/com/hello/visited/board/controller/BoardController.kt
package com.hello.visited.board.controller

import com.hello.visited.board.controller.response.GetBoardListResponse
import com.hello.visited.board.domain.BoardService
import com.hello.visited.board.domain.GetBoardResult
import com.hello.visited.response.DefaultResponse
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import javax.validation.Valid

@RestController
@RequestMapping("/board")
class BoardController(
    private val boardService: BoardService
) {

    @PostMapping
    fun makeBoard(@Valid @RequestBody request: MakeBoardRequest): DefaultResponse<Nothing> {
        boardService.makeBoard(request.toMakeBoardSource(), request.toUserSource())
        return DefaultResponse.success()
    }

    @GetMapping("/list")
    fun getBoards(): DefaultResponse<GetBoardListResponse> {
        val result = boardService.getBoards()
        return DefaultResponse.success(GetBoardListResponse.create(result))
    }

    @GetMapping
    fun getBoard(@Valid request: GetBoardResult): DefaultResponse<Nothing> {

        return DefaultResponse.success()
    }
}
